beacon_node/network/src/sync/message_processor.rs

```diff
@@ -438,6 +438,27 @@ impl<T: BeaconChainTypes> MessageProcessor<T> {
                 BlockProcessingOutcome::Processed { .. } => {
                     trace!(self.log, "Gossipsub block processed";
                             "peer_id" => format!("{:?}",peer_id));
+
+                    // TODO: It would be better if we can run this _after_ we publish the block to
+                    // reduce block propagation latency.
+                    //
+                    // The `MessageHandler` would be the place to put this, however it doesn't seem
+                    // to have a reference to the `BeaconChain`. I will leave this for future
+                    // works.
```

Comment:
Yeah agreed. The threading will be refactored in the network re-write. Will deal with this then